docs/release_process.md

```diff
@@ -0,0 +1,32 @@
+# cellxgene Release Process
```

Comment:
Minor nit. With the lowercase cellxgene it looks weird to have capitalized words after it. Can we do "Release Process for cellxgene" instead?

Reply:
or `#cellxgene release process` :) (that's my personal preference)

Reply:
Jeremy and lowercase... I'm fine with it either way

Reply:
resolved